File: 1_7/1_7.java
import java.util.Arrays;

public class Main {
    public static void main(String[] args) {
        String S = "1001";
        int x = 1;
        long c = Arrays.stream(S.split(""))
                      .filter(ch -> ch.equals("1"))
                      .count();
        boolean flag = true;
        int N = S.length();
        int limit = x * (int) c + 1;
        int max = 0;
        int count = 0;
        
        for (int i = 0; i < N - 1; i++) {
            if (S.charAt(i) == '0') {
                count += 1;
                if (max < count) {
                    max = count;
                } else {
                    count = 0;
                }
            }
            if (max >= limit) {
                flag = false;
                break;
            }
        }
        
        if (flag) {
            System.out.println("Yes");
        } else {
            System.out.println("No");
        }
    }
}

File: 1_7/1_7.js
const S = "1001";
let x = 1;
const c = S.split('').filter(char => char === '1').length;
let flag = true;
const N = S.length;
const limit = x * c + 1;
let max = 0;
let count = 0;

for (let i = 0; i < N - 1; i++) {
    if (S[i] === '0') {
        count += 1;
        if (max < count) {
            max = count;
        } else {
            count = 0;Ы
        }
    }
    if (max >= limit) {
        flag = false;
        break;
    }
}

if (flag) {
    console.log("Yes");
} else {
    console.log("No");
}